File: Step 06 LinkedList/Step 6.1 1D Linked List/Intro to LL/Intro to LL.cpp
// User function Template for C++

/*class Node {
public:
    int data;
    Node* next;
  
    // Default constructor
    Node()
    {
        data = 0;
        next = NULL;
    }
  
    // Parameterised Constructor
    Node(int data)
    {
        this->data = data;
        this->next = NULL;
    }
};*/

class Solution {
  public:
    Node* constructLL(vector<int>& arr) {
        // code here
        Node *head = new Node();
        Node *curr = head;
        
        for(auto num : arr){
            curr->next = new Node(num);
            curr = curr->next;
        }
        
        return head->next;
    }
};

File: Step 06 LinkedList/Step 6.1 1D Linked List/Intro to LL/Intro to LL.java


// User function Template for Java

/*
class Node { 
    int data; 
    Node next; 

    Node() { data = 0; }
    Node(int d) { data = d; }  //constructor to create a new node
} 
*/
class Solution {
    static Node constructLL(int arr[]) {
        // code here
        int len = arr.length;
        Node head=null, temp=null;
        int i = 0;
        while( i<len ){
            Node ptr = new Node(arr[i]);
            if(head==null) head = ptr;
            else temp.next = ptr;
            temp = ptr;
            i++;
        }
        return head;
    }
}